### PR TITLE
feat(issue-998): add validation for supported and missing data types in FIMSFrame

### PR DESCRIPTION
**Pull request by awilnoaa**
# What is the feature?

* Stops the model if any unsupported data types are present (e.g., “indexes” instead of “index”)
* Warns the user (but continues) if expected data types are missing (e.g., length_comp)

addresses issue #998 

# How have you implemented the solution?
* Defined which data types are allowed and/or expected
* Added a check for unsupported tops and stops execution if any are found
* Added a check for missing expected types and issues a warning instead of stopping execution
* Added three test_that() blocks to verify each scenario (error, warning, valid inputs)

# Does the PR impact any other area of the project, maybe another repo?
* No, changes should be contained within FIMSFrame

### CODE REVIEW

File: R/fimsframe.R [hunk 1]
```diff
@@ -569,7 +569,35 @@ methods::setValidity(
     }
 
     # TODO: Add checks for other slots
+    # Add validity check for types
+    allowed_types <- c(
+  "landings", "index", "age_comp", "length_comp",
+  "weight-at-age", "age-to-length-conversion"
+)
+present_types <- unique(object@data$type)
```
Comment (Bai-Li-NOAA) [inline R/fimsframe.R]: Could you change `object@data$type` to `object@data[["type"]]`? In FIMS, we use double brackets with quoted names instead of the dollar sign to avoid partial matching.

File: tests/testthat/test-fimsframe.R [hunk 2]
```diff
@@ -208,4 +208,33 @@ test_that("get_n_fleets() works with correct inputs", {
 # No edge cases to test.
 
 ## Error handling ----
-# No built-in errors to test.
+
+## Check that user does not include unsupported data types
+test_that("FIMSFrame() aborts on unsupported data types", {
+  data("data1", package = "FIMS")
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-fimsframe.R]: We can remove this line and use `data1` directly.

File: tests/testthat/test-fimsframe.R [hunk 3]
```diff
@@ -208,4 +208,33 @@ test_that("get_n_fleets() works with correct inputs", {
...
+  bad <- dplyr::mutate(
+    data1,
+    type = ifelse(type == "index", "indexes", type)  # Introduce an unsupported type
+  )
+  expect_error(FIMS::FIMSFrame(bad), regexp = "unsupported type\\(s\\)")
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-fimsframe.R]: Could you add a `@description` tag with a **one-line** summary of the test above `expect_error()` so it can be rendered in future {testdown} reports? Please see the example at line 191.

File: tests/testthat/test-fimsframe.R [hunk 4]
```diff
@@ -208,4 +208,33 @@ test_that("get_n_fleets() works with correct inputs", {
...
+
+  expect_warning(
+    ff <- FIMS::FIMSFrame(no_len),
+    regexp = "expected data type\\(s\\) are not present"
+  )
+  expect_s4_class(ff, "FIMSFrame")
+})
+
+##Confirm that valid inputs work without error
+test_that("FIMSFrame() succeeds cleanly with valid inputs", {
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-fimsframe.R]: Thanks for writing this test! Do you think we could move it to the IO correctness section, maybe around line 25?

**Review by Bai-Li-NOAA — CHANGES_REQUESTED** [review]
Thanks for tackling this issue! It looks really good. I have a few small notes on documentation and style, and one comment on the warning message. Let me know if anything’s unclear. Feel free to push more commits to the feature branch and squash them into one at the end. Happy to demo squashing using the VS Code terminal if needed.

File: tests/testthat/test-fimsframe.R (hunk 3 above)
Comment (awilnoaa) [reply tests/testthat/test-fimsframe.R]: Ok, added.

File: tests/testthat/test-fimsframe.R [hunk 5]
```diff
@@ -208,4 +208,33 @@ test_that("get_n_fleets() works with correct inputs", {
...
+  data("data1", package = "FIMS")
```
[the remark this answers is not in the document]
Comment (awilnoaa) [reply tests/testthat/test-fimsframe.R]: Test was moved to IO correctness section.
